### PR TITLE
Add gas payments to kathy

### PR DESCRIPTION
**Pull request by asaj**
Switches Kathy to estimatePaymentForHandleGas.

### CODE REVIEW

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 1]
```diff
@@ -66,13 +69,27 @@ async function main() {
   }, 1000 * 30);
 
   for (const origin of origins) {
+    console.log(origins);
```
Comment (tkporter) [inline typescript/infra/scripts/helloworld/kathy.ts:72]: rm?

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 2]
```diff
@@ -66,13 +69,27 @@ async function main() {
...
     for (const destination of origins.filter((d) => d !== origin)) {
       const labels = {
         origin,
         remote: destination,
         ...constMetricLabels,
       };
       try {
+        const actual = await app
```
Comment (tkporter) [inline typescript/infra/scripts/helloworld/kathy.ts:80]: is this intentional to keep in here?

Comment (asaj) [reply typescript/infra/scripts/helloworld/kathy.ts:80]: Good catch, forgot I added this.

File: typescript/helloworld/src/app/app.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export class HelloWorldApp<
     const estimated = await sender.estimateGas.sendHelloWorld(
       toDomain,
       message,
-      chainConnection.overrides,
+      { ...chainConnection.overrides, value },
```
Comment (yorhodes) [inline typescript/helloworld/src/app/app.ts:33]: good catch

File: typescript/sdk/src/gas/calculator.ts [hunk 4]
```diff
@@ -289,7 +290,7 @@ export class InterchainGasCalculator<Chain extends ChainName> {
       from: destinationInbox.address,
       data: handlerInterface.encodeFunctionData('handle', [
         chainMetadata[message.origin].id,
-        message.sender,
+        utils.addressToBytes32(message.sender),
```
Comment (yorhodes) [inline typescript/sdk/src/gas/calculator.ts:293]: how did this work previously?

**Review by yorhodes — APPROVED** [review]

File: typescript/sdk/src/gas/calculator.ts (hunk 4 above)
Comment (asaj) [reply typescript/sdk/src/gas/calculator.ts:293]: It did not

**Review by tkporter — APPROVED** [review]
btw https://github.com/abacus-network/abacus-monorepo/pull/806#discussion_r932100318

**Pull request merged**